service/history/execution/mutable_state_builder_test.go

```diff
@@ -1959,3 +1959,226 @@ func TestMutableStateBuilder_CopyToPersistence_roundtrip(t *testing.T) {
 
 	}
 }
+
+func TestMutableStateBuilder_closeTransactionHandleWorkflowReset(t *testing.T) {
+
+	t1 := time.Unix(123, 0)
+	now := time.Unix(500, 0)
+
+	badBinaryID := "bad-binary-id"
+
+	mockDomainEntryWithBadBinary := cache.NewLocalDomainCacheEntryForTest(&persistence.DomainInfo{Name: "domain"}, &persistence.DomainConfig{
+		BadBinaries: types.BadBinaries{
+			Binaries: map[string]*types.BadBinaryInfo{
+				badBinaryID: &types.BadBinaryInfo{
+					Reason:          "some-reason",
+					Operator:        "",
+					CreatedTimeNano: common.Ptr(t1.UnixNano()),
+				},
+			},
+		},
+	}, "cluster0")
+
+	mockDomainEntryWithoutBadBinary := cache.NewLocalDomainCacheEntryForTest(nil, &persistence.DomainConfig{
+		BadBinaries: types.BadBinaries{
+			Binaries: map[string]*types.BadBinaryInfo{},
+		},
+	}, "cluster0")
+
+	tests := map[string]struct {
+		policyIn                         TransactionPolicy
```

Comment:
`TransactionPolicyPassive` not covered by any test cases

Reply:
true, I forgot, too much had time since I wrote this. let me add